File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i = 0; i < n; i++) {
            String binary1 = Integer.toBinaryString(arr1[i]);
            String binary2 = Integer.toBinaryString(arr2[i]);
            binary1 = String.format("%" + n + "s", binary1).replace(" ", "0");
            binary2 = String.format("%" + n + "s", binary2).replace(" ", "0");
            
            String row = "";
            for (int j = 0; j < n; j++) {
                if (binary1.charAt(j) == '0' && binary2.charAt(j) == '0') {
                    row += " ";
                } else {
                    row += "#";
                }
            }
            
            answer[i] = row;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(n):
        a = bin(arr1[i])[2:].zfill(n)
        b = bin(arr2[i])[2:].zfill(n)
        row = ''
        for j in range(n):
            if a[j] == '0' and b[j] == '0':
                row += ' '
            else:
                row += '#'
        answer.append(row)
    
    return answer
